addons/website/controllers/main.py

```diff
@@ -236,10 +236,24 @@ def create_sitemap(url, content):
 
         return request.make_response(content, [('Content-Type', mimetype)])
 
-    @http.route('/website/info', type='http', auth="public", website=True, sitemap=True)
+    def sitemap_website_info(env, rule, qs):
+        website = env['website'].get_current_website()
+        if not (
+            website.viewref('website.website_info', False).active
+            and website.viewref('website.show_website_info', False).active
+        ):
+            # avoid 404 or blank page in sitemap
+            return False
+
+        if not qs or qs.lower() in '/website/info':
+            yield {'loc': '/website/info'}
+
+    @http.route('/website/info', type='http', auth="public", website=True, sitemap=sitemap_website_info)
     def website_info(self, **kwargs):
         if not request.website.viewref('website.website_info', False).active:
-            # delete or archived view
+            # Deleted or archived view (through manual operation in backend).
+            # Don't check `show_website_info` view: still need to access if
+            # disabled to be able to enable it through the customize show.
```

Comment:
Oh, ok, you did it in the next commit maybe this whole comment should be in the first commit?

Reply:
Could be indeed, even if the goal was to make it clear why in this commit the sitemap method is copying the check from the controller but also checking for the other view too while here we shouldn't.
I splitted the comment in the two commits :+1: